### PR TITLE
feat/FWS-98/Add_email_notifications

### PR DESCRIPTION
**Pull request by Eli293**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-98
## Context:
This ticket is about the addition of email notifications for more than just the appointment cancellation, it also includes the use of templates for the emails. 
 
## Changes

- New emails notifications (look)
- New emails triggers
- Moved emails from frontend to backend
- No longer using an API to send emails
- Affects all roles

## NEW EMAILS
NEW WELCOME EMAIL
image
NEW APPOINTMENT CANCELLED EMAIL
image
NEW APPOINTMENT CONFIRMATION EMAIL
image

### CODE REVIEW

File: fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js [hunk 1]
```diff
@@ -178,75 +178,7 @@ function AdminAccounts() {
...
-                return response.text();;
-            })
-            .then((data) => {
-                console.log(data);
-            })
-            .catch((error) => {
-                console.log(error);
-
-            });
-    };
```
Comment (emilegirardGit) [inline fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js:249]: Why did you delete everything here?

**Review by emilegirardGit — APPROVED** [review]

File: fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js (hunk 1 above)
Comment (Eli293) [reply fitwsarah_frontend/src/views/PersonalTrainerPanel/TrainerAppointments.js:249]: Good question! I removed them because I moved the logic behind javaMail to the backend only instead of the frontend to avoid redundant code clustering in the frontend code.

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ services:
       - SPRING_PROFILES_ACTIVE=dev
       - AUTH0_AUDIENCE=${AUTH0_AUDIENCE}
       - AUTH0_ISSUER=${AUTH0_ISSUER}
+      - MAIL_PASSWORD=${MAIL_PASSWORD}
```
Comment (AlexeiTimbro) [inline docker-compose.dev.yml:11]: could you elaborate on why you added this environment variable? and what it does.

**Review by AlexeiTimbro — APPROVED** [review]
I added a comment answer it before you merged good job elias.

File: docker-compose.dev.yml (hunk 2 above)
Comment (Eli293) [reply docker-compose.dev.yml:11]: This is the mail password for accessing the gmail server to send emails.
